web_src/js/features/repo-projects.js

```diff
@@ -168,24 +182,27 @@ export function initRepoProject() {
     });
   });
 
-  $('#new_board_submit').on('click', function (e) {
+  $('#new_board_submit').on('click', (e) => {
     e.preventDefault();
-
     const boardTitle = $('#new_board');
     const projectColorInput = $('#new_board_color_picker');
+    if (boardTitle.val().length < 1) {
+      return false;
+    }
+    createNewBoard(boardTitle, projectColorInput);
+  });
 
-    $.ajax({
-      url: $(this).data('url'),
-      data: JSON.stringify({title: boardTitle.val(), color: projectColorInput.val()}),
-      headers: {
-        'X-Csrf-Token': csrfToken,
-      },
-      contentType: 'application/json',
-      method: 'POST',
-    }).done(() => {
-      boardTitle.closest('form').removeClass('dirty');
-      window.location.reload();
-    });
+  $('.new-board').on('keyup', (e) => {
+    const boardTitle = $('#new_board');
+    const projectColorInput = $('#new_board_color_picker');
+    if (boardTitle.val().length < 1) {
+      $('#new_board_submit').addClass('disabled');
+      return false;
```

Comment:
Do not `return false` in keyboard event listener.

Reply:
I am not using e.preventDefault in `key` event handler. Shall I.

Reply:
No need to prevent anything in input event listener.